xbmc/listproviders/DirectoryProvider.cpp

```diff
@@ -272,7 +274,16 @@ void CDirectoryProvider::OnEvent(const ADDON::AddonEvent& event)
         typeid(event) == typeid(ADDON::AddonEvents::Disabled) ||
         typeid(event) == typeid(ADDON::AddonEvents::InstalledChanged) ||
         typeid(event) == typeid(ADDON::AddonEvents::MetadataChanged))
-    m_updateState = INVALIDATED;
+      m_updateState = INVALIDATED;
+  }
+}
+
+void CDirectoryProvider::Notify(const Observable &obs, const ObservableMessage msg)
+{
+  if (URIUtils::IsProtocol(m_currentUrl, "pvr"))
+  {
+    if (msg == ObservableMessageManagerStateChanged && g_PVRManager.IsStarted())
```

Comment:
I understand what you mean and your intention for sure is okay. But changing pvr messaging is high effort and i do not want to do this **now** as we are near beta 1 and this is highly sensitive pvr code (I never touched/read before).  As i wrote i will be happy to (completely) rework this during alpha phase for v18.